internal/dependency/manager.go

```diff
@@ -152,147 +89,102 @@ func (dm *DependendcyManager) Validate(ctx context.Context, piManifest *v1alpha1
 	}, nil
 }
 
-func (dm *DependendcyManager) CheckConflict(existingVersionStr string, dependency v1alpha1.Dependency) (*Conflict, error) {
-	existingVersion, err := semver.NewVersion(existingVersionStr)
-	if err != nil {
-		return nil, err
-	}
-	requiredRange, err := semver.NewConstraint(dependency.Version)
+// NewGraph constructs a DependencyGraph from all packages returned by clientAdapter.ListPackages
+func (dm *DependendcyManager) NewGraph(ctx context.Context) (*graph.DependencyGraph, error) {
+	pkgs, err := dm.clientAdapter.ListPackages(ctx)
 	if err != nil {
 		return nil, err
 	}
-	if ok := requiredRange.Check(existingVersion); !ok {
-		return &Conflict{
-			Actual: PackageWithVersion{
-				Name:    dependency.Name,
-				Version: existingVersionStr,
-			},
-			Required: PackageWithVersion{
-				Name:    dependency.Name,
-				Version: dependency.Version,
-			},
-		}, nil
-	} else {
-		return nil, nil
-	}
-}
-
-func (dm *DependendcyManager) createRequirement(dependency v1alpha1.Dependency) (*PackageWithVersion, error) {
-	requirement := &PackageWithVersion{
-		Name: dependency.Name,
-	}
-	if dependency.Version == "" {
-		if latest, err := dm.repoAdapter.GetLatestVersion("", dependency.Name); err != nil {
-			return nil, err
+	g := graph.NewGraph()
+	for _, pkg := range pkgs.Items {
+		var deps []v1alpha1.Dependency
+		installedVersion := pkg.Spec.PackageInfo.Version
+		if !pkg.DeletionTimestamp.IsZero() {
+			// A package that is currently being deleted is added to the graph, but in a state representing
+			// "not installed"
+			installedVersion = ""
 		} else {
-			requirement.Version = latest
+			if pi, err := dm.clientAdapter.GetPackageInfo(ctx, pkg.Status.OwnedPackageInfos[0].Name); err != nil {
```

Comment:
and more importantly, I have two cases where I run into a panic in this line, because `OwnedPackageInfos` is `nil`. 

1. When installing a package via the UI, immediately afterwards the updatability is being checked and then broadcasted via websocket. Maybe the package status does not contain this information yet directly after the package has been added?
2. When installing a package with a previously manually installed dependency via the CLI, this line panics in the package controller for the same reason. I don't know though if it makes a difference whether the package has a dependency or not. Maybe it was just a coincidence.